src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -665,13 +672,13 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                         Error(diagnostics, ErrorCode.ERR_AnonymousTypeDuplicatePropertyName, argumentSyntax.NameColon.Name);
```

Comment:
It does not seem likely that both errors will be reported. It would need to be both common and duplicate name.
The other point for reporting separate errors could be if we decide to add more "reserved" names like "ToString" - not quite sure about that yet though.